acceptance-tests/src/test/java/it/com/atlassian/bitbucket/jenkins/internal/SmokeTest.java

```diff
@@ -0,0 +1,444 @@
+import org.jenkinsci.test.acceptance.po.Build;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+import org.jenkinsci.test.acceptance.po.JenkinsConfig;
+import org.jenkinsci.test.acceptance.po.WorkflowJob;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.net.URLEncoder;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static io.restassured.http.ContentType.JSON;
+import static it.com.atlassian.bitbucket.jenkins.internal.util.BitbucketUtils.*;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.time.Duration.ofMinutes;
+import static java.util.UUID.randomUUID;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.jenkinsci.test.acceptance.plugins.credentials.ManagedCredentials.DEFAULT_DOMAIN;
+import static org.jenkinsci.test.acceptance.po.Build.Result.SUCCESS;
+
+@WithPlugins({"mailer", "atlassian-bitbucket-server-integration"})
+public class SmokeTest extends AbstractJUnitTest {
+
+    @Rule
+    public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private String bbsAdminCredsId;
+    private BitbucketRepository forkRepo;
+    private String serverId;
+
+    @Before
+    public void setUp() {
+        // BB Access Token
+        PersonalToken bbsToken = createPersonalToken(PROJECT_READ_PERMISSION, REPO_ADMIN_PERMISSION);
+        CredentialsPage credentials = new CredentialsPage(jenkins, DEFAULT_DOMAIN);
+        credentials.open();
+        BitbucketTokenCredentials bbsTokenCreds = credentials.add(BitbucketTokenCredentials.class);
+        bbsTokenCreds.token.set(bbsToken.getSecret());
+        bbsTokenCreds.description.sendKeys(bbsToken.getId());
+        credentials.create();
+
+        // BB Admin user/password
+        credentials.open();
+        UserPwdCredential bbAdminCreds = credentials.add(UserPwdCredential.class);
+        bbsAdminCredsId = "admin-" + randomUUID();
+        bbAdminCreds.setId(bbsAdminCredsId);
+        bbAdminCreds.username.set(BITBUCKET_ADMIN_USERNAME);
+        bbAdminCreds.password.set(BITBUCKET_ADMIN_PASSWORD);
+        credentials.create();
+
+        // Create BB Server entry
+        JenkinsConfig jenkinsConfig = jenkins.getConfigPage();
+        jenkinsConfig.configure();
+        serverId = "bbs-" + randomUUID();
+        new BitbucketPluginConfiguration(jenkinsConfig)
+                .addBitbucketServer(serverId, BITBUCKET_BASE_URL, bbsToken.getId(), bbsAdminCredsId);
+        jenkinsConfig.save();
+
+        // Fork repo
+        forkRepo = forkRepository(PROJECT_KEY, REPO_SLUG, "fork-" + randomUUID());
+    }
+
+    @Test
+    public void testFreeStyleJob() throws IOException, GitAPIException {
```

Comment:
I wouldn't mind renaming these method to long names to convey what the test is doing. This is an end to end test and would help identify problem by looking into the test failed. For example [UnitOfWork_StateUnderTest_ExpectedBehavior], as explained in top voted answer here - https://stackoverflow.com/questions/155436/unit-test-naming-best-practices

Reply:
- [ ] rename test methods